charts/wildfly/templates/route.yaml

```diff
@@ -1,3 +1,6 @@
+{{- if (and .Values.deploy.route.enabled .Values.deploy.ingress.enabled) }}
```

Comment:
maybe we could be smarter here.

The `route.openshift.io/v1` kind does not exist on Kubernetes (only on OpenShift).
Maybe we could look up if  `route.openshift.io/v1`  exists to create the route.

If `route.openshift.io/v1`  exists, then we fail and let the user know they will have to disable one of them.

For plain kubernetes deployment, this will simplify the configuration with:

```
deploy:
  ingress:
    enabled: true
```

without having to disable `route` explicitly.

For OpenShift deployment, the user will have to decide between

```
deploy:
  ingress:
    enabled: true
  route:
    enabled: false
```

or the default config (route enabled, ingress disabled).

wdyt?

Reply:
that's a nice improvement

Reply:
I'm wondering about creating ingress resource on Openshift. What would be the point  ?